packages/iov-core/src/jsonrpcsigningserver.ts

```diff
@@ -0,0 +1,184 @@
+import { ChainId, UnsignedTransaction } from "@iov/bcp-types";
+import {
+  isJsonCompatibleDictionary,
+  jsonRpcCodeInvalidParams,
+  jsonRpcCodeInvalidRequest,
+  jsonRpcCodeMethodNotFound,
+  jsonRpcCodeServerErrorDefault,
+  JsonRpcErrorResponse,
+  JsonRpcRequest,
+  JsonRpcResponse,
+  parseJsonRpcId,
+  parseJsonRpcRequest,
+} from "@iov/jsonrpc";
+
+import { SigningServerCore } from "./signingservercore";
+
+interface RpcCallGetIdentities {
+  readonly name: "getIdentities";
+  readonly reason: string;
+  readonly chainIds: ReadonlyArray<ChainId>;
+}
+
+interface RpcCallSignAndPost {
+  readonly name: "signAndPost";
+  readonly reason: string;
+  readonly transaction: UnsignedTransaction;
+}
+
+type RpcCall = RpcCallGetIdentities | RpcCallSignAndPost;
+
+class ParamsError extends Error {}
+class MethodNotFoundError extends Error {}
+
+function isArrayOfStrings(array: ReadonlyArray<any>): array is ReadonlyArray<string> {
+  return array.every(element => typeof element === "string");
+}
+
+function parseRpcCall(data: JsonRpcRequest): RpcCall {
+  if (!isJsonCompatibleDictionary(data.params)) {
+    throw new Error("Request params are only supported as dictionary");
+  }
+
+  switch (data.method) {
+    case "getIdentities": {
+      const reason = data.params.reason;
+      const chainIds = data.params.chainIds;
+      if (typeof reason !== "string") {
+        throw new ParamsError("1st parameter (reason) must be a string");
+      }
+      if (!Array.isArray(chainIds)) {
+        throw new ParamsError("2nd parameter (chainIds) must be an array");
+      }
+      if (!isArrayOfStrings(chainIds)) {
+        throw new ParamsError("Found non-string element in chainIds array");
+      }
+      return {
+        name: "getIdentities",
+        reason: reason,
+        chainIds: chainIds,
+      };
+    }
+    case "signAndPost": {
+      const reason = data.params.reason;
+      const transaction = data.params.transaction;
+      if (typeof reason !== "string") {
+        throw new ParamsError("1st parameter (reason) must be a string");
+      }
+      if (typeof transaction !== "object") {
```

Comment:
Not more tests than Object?
What about asserting the common fields exist at least, like kind, signer, etc....

Reply:
Added some more: https://github.com/iov-one/iov-core/pull/654/commits/084c472c35f4efbb720efa25c3c06e2c39448c0f